server/src/main/java/io/druid/segment/loading/SegmentLoaderLocalCacheManager.java

```diff
@@ -175,11 +193,17 @@ public void cleanup(DataSegment segment) throws SegmentLoadingException
     }
 
     try {
-      // Druid creates folders of the form dataSource/interval/version/partitionNum.
-      // We need to clean up all these directories if they are all empty.
-      File cacheFile = new File(loc.getPath(), DataSegmentPusherUtil.getStorageDir(segment));
-      cleanupCacheFiles(loc.getPath(), cacheFile);
-      loc.removeSegment(segment);
+      // iterator all locations in case segment stored in more than one location due to the retry in getSegmentFiles()
```

Comment:
If storageDir.mkdirs() success, but downloadStartMarker.createNewFile() failed, in this case, findStorageLocationIfLoaded() will think segment is located in the bad storageDir which is actually not.
So we should always remove all possible locations when cleanup